src/pin_layout.rs

```diff
@@ -243,63 +251,73 @@ fn create_pin_view_side(
                 Some(other)
             }
         };
+
         pin_options_row = pin_options_row.push(
             pick_list(config_options, selected, move |pin_function| {
                 Message::PinFunctionSelected(board_pin_number, bcm_pin_number, pin_function)
             })
-            .width(Length::Fixed(200f32))
+            .width(Length::Fill)
             .placeholder("Select function"),
         );
 
         pin_option = pin_option.push(pin_options_row);
     }
 
+    let mut pin_name_column = Column::new()
+        .width(Length::Fixed(70f32))
+        .align_items(Alignment::Center);
+
     // Create the Pin name
     let pin_name = Row::new()
-        .width(Length::Fixed(55f32))
         .push(Text::new(pin_description.name))
         .align_items(Alignment::Center);
 
-    let mut pin_arrow = Row::new()
-        .width(Length::Fixed(60f32))
-        .align_items(Alignment::Center);
+    pin_name_column = pin_name_column.push(pin_name);
+
+    let mut pin_arrow_column = Column::new()
+        .align_items(Alignment::Center)
+        .width(Length::Fixed(30f32));
+
+    let mut pin_arrow = Row::new().align_items(Alignment::Center);
+
     if is_left {
         pin_arrow = pin_arrow.push(circle(5.0));
-        pin_arrow = pin_arrow.push(line(50.0));
+        pin_arrow = pin_arrow.push(line(20.0));
     } else {
-        pin_arrow = pin_arrow.push(line(50.0));
+        pin_arrow = pin_arrow.push(line(20.0));
         pin_arrow = pin_arrow.push(circle(5.0));
     }
 
+    pin_arrow_column = pin_arrow_column.push(pin_arrow);
+
+    let mut pin_button_column = Column::new().align_items(Alignment::Center);
     // Create the pin itself, with number and as a button
     let pin_button = button(
         Text::new(pin_description.board_pin_number.to_string())
-            .size(20)
             .horizontal_alignment(Horizontal::Center),
     )
-    .padding(10)
-    .width(Length::Fixed(40f32))
+    .padding(5)
+    .width(Length::Fixed(30f32))
     .style(get_pin_color(pin_description).get_button_style())
     .on_press(Message::Activate(pin_description.board_pin_number));
 
+    pin_button_column = pin_button_column.push(pin_button);
```

Comment:
Do you need the extra column around the button?

Reply:
yaa, otherwise the layout gets disurbed